blockprod/src/lib.rs

```diff
@@ -48,6 +51,10 @@ pub enum BlockProductionError {
     FailedConsensusInitialization(#[from] ConsensusCreationError),
     #[error("Block production cancelled")]
     Cancelled,
+    #[error("Failed to retieve peer count")]
+    PeerCountRetrievalError,
+    #[error("Connected peers is belowe the required peer threshold")]
+    PeerCountBelowRequiredThreshold,
```

Comment:
Please include the required threshold and the current value and print them in the error string.